lib/Dialect/LLHD/Transforms/ProcessLoweringPass.cpp

```diff
@@ -38,31 +41,46 @@ static LogicalResult isProcValidToLower(llhd::ProcessOp op) {
   size_t numBlocks = op.getBody().getBlocks().size();
 
   if (numBlocks == 1) {
-    if (!isa<llhd::HaltOp>(op.getBody().back().getTerminator()))
-      return op.emitOpError("during process-lowering: entry block is required "
-                            "to be terminated by llhd.halt");
+    if (!isa<llhd::HaltOp>(op.getBody().back().getTerminator())) {
+      LLVM_DEBUG({
+        llvm::dbgs()
+            << "entry block is required to be terminated by llhd.halt\n";
+      });
+      return failure();
+    }
```

Comment:
Is it legal to inline a process with a `halt` terminator into the parent module? Wouldn't the process only execute once, but the inlined version would be continuously executed?

Reply:
That's right, let me remove this case. Not sure why it was there to begin with.